src/ptf/dataplane.py

```diff
@@ -27,13 +27,20 @@
 from threading import Thread
 from threading import Lock
 from threading import Condition
-import ptfutils
-import netutils
-import mask
+from . import ptfutils
+from . import netutils
+from . import mask
 import scapy.packet
 import scapy.utils
-from pcap_writer import PcapWriter
-from StringIO import StringIO
+from .pcap_writer import PcapWriter
+# Python3 doesn't have support for StringIO.StringIO
+# Python2 does have support for both but io.StringIO
+# only accepts unicode which is not the usual str code
+# used by print statements in python2.x
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
```

Comment:
Given other changes you've made (including the above imports), it seems to be that keeping compatibility with Python2 is broken anyway. So why try to maintain it here?
If you do want, I would recommend you explicitly use sys.version_info.major == 2 (or 3), rather than relying on exceptions.